packages/txservice/src/provider.ts

```diff
@@ -0,0 +1,177 @@
+      // which might be okay in our case, but for now we have a low bar.
+      this.provider = new FallbackProvider(fallbackProviderConfigs, 1);
+    }
+
+    this.signer = typeof signer === "string" ? new Wallet(signer, this.provider) : signer.connect(this.provider);
+  }
+
+  public async sendTransaction(
+    tx: FullTransaction
+  ): Promise<{ response: providers.TransactionResponse | Error; success: boolean }> {
+    const ready = await this.provider.ready;
+    if (!ready) {
+      // Error out, no providers are ready.
+      throw new ChainError(ChainError.reasons.ProviderNotReady);
+    }
+    // TODO: Check getBlockNumber() > 0.
+
+    // Define task to send tx with proper nonce
+    const task = async (): Promise<{ response: providers.TransactionResponse | Error; success: boolean }> => {
+      try {
+        // Send transaction using the passed in callback.
+        // const stored = this.nonces.get(chainId);
+        // const nonceToUse: number = nonce ?? stored ?? (await signer.getTransactionCount("pending"));
+        const pending: number = await this.signer.getTransactionCount("pending");
+        const { to, data, chainId, value, gasPrice, nonce } = tx;
+        const response: providers.TransactionResponse | undefined = await this.signer.sendTransaction({
+          to,
+          data,
+          chainId,
+          gasPrice,
+          nonce: nonce ?? pending,
+          value: BigNumber.from(value || 0),
+        });
+        // After calling tx fn, set nonce to the greatest of
+        // stored, pending, or incremented
+        // const pending = await signer.getTransactionCount("pending");
+        // const incremented = (response?.nonce ?? nonceToUse) + 1;
+        // Ensure the nonce you store is *always* the greatest of the values
+        // const toCompare = stored ?? 0;
+        // if (toCompare < pending || toCompare < incremented) {
+        //   this.nonces.set(chainId, incremented > pending ? incremented : pending);
+        // }
+        // Check to see if ethers returned undefined for the response; if so, handle as error case.
+        if (!response) {
+          throw new ChainError(ChainError.reasons.FailedToSendTx);
+        }
+        return { response, success: true };
+      } catch (e) {
+        return { response: e, success: false };
+      }
+    };
+    // Queue up the execution of the transaction.
+    return await this.queue.add(task);
+  }
+
+  public async confirmTransaction(
+    transactionHash: string,
+    timeout: number,
+  ) {
+    // We are using waitForTransaction here to leverage the timeout functionality internal to ethers.
+    // IF this times out, ethers will reject with ("timeout exceeded", Logger.errors.TIMEOUT)
+    // Alternatively, it could reject with ("transaction was replaced", Logger.errors.TRANSACTION_REPLACED)
+    await this.provider.waitForTransaction(transactionHash, this.confirmationsRequired, timeout);
```

Comment:
I believe this does return a `TransactionReceipt` so you should only need the one call